sensor/common/detector/filesystem/service/service_impl.go

```diff
@@ -0,0 +1,92 @@
+package service
+
+import (
+	"context"
+	"io"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/grpc/authz/idcheck"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/sensor/common/message"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// NewService creates a new streaming service with the fact agent. It should only be called once.
+func NewService() Service {
+	srv := &serviceImpl{
+		authFuncOverride: authFuncOverride,
```

Comment:
If you are not planning on overriding the `AuthFuncOverride` (typically for testing) you can just simply:
```golang
func (s *serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
	return ctx, errors.Wrapf(idcheck.CollectorOnly().Authorized(ctx, fullMethodName), "file activity authorization for  %q", fullMethodName)
}
```

Reply:
I am not sure if I did what you asked be to do.